src/compose/SendButton.js

```diff
@@ -30,14 +30,31 @@ export default class SendButton extends React.Component {
 
   render() {
     const { disabled, onPress } = this.props;
-    const opacity = { opacity: disabled ? 0.25 : 1 };
+    const opacity = { opacity: 0.25 };
 
-    return (
-      <Touchable style={styles.wrapper} onPress={disabled ? undefined : onPress}>
-        <View style={[styles.button, opacity]}>
-          <IconSend size={16} color="white" />
+    /*
+      Since ReactNative's communication with the native mobile component is through a bridge that is
+      not very fast, propagation of change in state from the parent controlled input 
+      (ComposeText.js) results in button remaining in highlighted state event after the button is no
+      longer being tapped. To solve this we remove <Touchable/> wrapping when the button is in 
+      disabled state.
+    */
+    if (disabled) {
```

Comment:
We can express this simpler and with less repetition by:

```js
const WrapperComponent = disabled ? View : Touchable;
<WrapperComponent .... />
```

Even better, we should add this code to our Touchable component, because we likely we'll want to reuse it.